src/explore-education-statistics-frontend/src/pages/api/permalink.ts

```diff
@@ -0,0 +1,72 @@
+import mapFullTable from '@common/modules/table-tool/utils/mapFullTable';
+import mapTableHeadersConfig from '@common/modules/table-tool/utils/mapTableHeadersConfig';
+import mapTableToJson from '@common/modules/table-tool/utils/mapTableToJson';
+import generateTableTitle from '@common/modules/table-tool/utils/generateTableTitle';
+import type { NextApiRequest, NextApiResponse } from 'next';
+
+interface ResponseData {
+  table?: string;
+  title?: string;
+  error?: unknown;
+}
+
+/**
+ * Endpoint to generate table json and title for permalinks.
+ *
+ * Expects req.body to be:
+ * {
+ *  fullTable: TableDataResponse,
+ *  configuration: {
+ *   tableHeaders: UnmappedTableHeadersConfig
+ *  }
+ * }
+ */
+export default async function handlePermalink(
+  req: NextApiRequest,
+  res: NextApiResponse<ResponseData>,
+) {
+  if (
+    process.env.NODE_ENV === 'production' &&
+    req.query.PERMALINK_API_SECRET !== process.env.PERMALINK_API_SECRET
+  ) {
+    res.status(401).send({ error: 'Unauthorised' });
```

Comment:
We can make this a bit more succulent and just do `return res.status(401).send({ error: 'Unauthorised' });`